pandas/io/parsers.py

```diff
@@ -2346,9 +2346,17 @@ def __init__(self, f: Union[FilePathOrBuffer, List], **kwds):
             raise ValueError("Only length-1 decimal markers supported")
 
         if self.thousands is None:
-            self.nonnum = re.compile(fr"[^-^0-9^{self.decimal}]+")
+            regex = fr"^\-?[0-9]*({self.decimal}[0-9]*)?([0-9](E|e)\-?[0-9]*)?$"
```

Comment:
is self.decimal already escaped by the time we get here?

Reply:
No, seems not to be the case. Escaped it.